### PR TITLE
feat: ADDON-61556 Introduced checkbox group component

### PR DESCRIPTION
**Pull request by tbalar-splunk**
**Jira -** https://splunk.atlassian.net/browse/ADDON-61556

Frontend changes has been done.

Backend Changes Jira - https://splunk.atlassian.net/browse/ADDON-64823
Doc Changes Jira - https://splunk.atlassian.net/browse/ADDON-64824

We've introduced a new entity type named **checkboxGroups** along with the addition of the **CheckboxGroupsMetadata** property. This new property allows for the utilization of this entity in a grouped format.

The following screenshot presents a global config snippet that demonstrates how to utilize checkbox group.

checkbox-groups

> The detailed documentation will be done as a separate task.

**Checkbox With Groups Feature:**
Screenshot 2023-08-28 at 12 08 47 PM

**Without Grouping Feature: (If 'groups' property is not specified inside the 'CheckboxGroupsMetadata' property)**
Screenshot 2023-08-28 at 12 09 31 PM

**This is how payload will pass to the backend:**
Screenshot 2023-08-28 at 12 12 58 PM

### CODE REVIEW

File: src/main/webapp/components/BaseFormView.jsx [hunk 1]
```diff
@@ -99,6 +98,29 @@ class BaseFormView extends PureComponent {
                     }
                     if (props.mode === MODE_EDIT || props.mode === MODE_CLONE) {
                         this.currentInput = context.rowData[props.serviceName][props.stanzaName];
+
+                        // Convert a comma-separated string of key-value pairs into a dictionary.
```
[the remark this answers is not in the document]
Comment (vtsvetkov-splunk) [reply src/main/webapp/components/BaseFormView.jsx:102]: Why should we wait a perfect moment for big refactoring if we can make a small step now?

File: src/main/webapp/components/CheckboxGroupsComponent.jsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    const { field, label, value, checkboxTextFieldValue, handleChange } = props;
+
+    const [isDisabled, setIsDisabled] = useState(!value);
+
+    useEffect(() => {
+        setIsDisabled(!value);
+    }, [value]);
+
+    const handleChangeCheckbox = () => {
+        if (value && !isFalse(value)) {
```
Comment (wmazur-splunk) [inline src/main/webapp/components/CheckboxGroupsComponent.jsx:18]: why didn't you use isTrue() ?

**Review by wmazur-splunk — APPROVED** [review]
No tests :-(

File: src/main/webapp/components/CheckboxGroupsComponent.jsx (hunk 2 above)
Comment (tbalar-splunk) [reply src/main/webapp/components/CheckboxGroupsComponent.jsx:18]: Nice catch!! Let me use isTrue and will make sure that everything is working fine.

**Comment by tbalar-splunk** [comment]
> No tests :-(

Since we planned to release this feature in August I didn’t make it scoped. The test tasks should be in the backlog, let me check. (If not, I will will make) )

File: src/main/webapp/components/BaseFormView.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (tbalar-splunk) [reply src/main/webapp/components/BaseFormView.jsx]: Did necessary changes

**Review by wmazur-splunk — APPROVED** [review]

**Comment by artemrys** [comment]
I would wait for @vtsvetkov-splunk approval before merging it.

**Comment by tbalar-splunk** [comment]
FYI - Actions are not working

**Pull request merged**